src/sparsematrix.jl

```diff
@@ -931,6 +931,31 @@ function Matrix(S::AbstractSparseMatrixCSC{Tv}) where Tv
 end
 Array(S::AbstractSparseMatrixCSC) = Matrix(S)
 
+function Base.copyto!(A::Array{T}, S::SparseMatrixCSC{<:Number}) where {T<:Number}
+    isempty(S) && return A
+    length(A) < length(S) && throw(BoundsError())
+    
+    # Zero elements that are also in S, don't change rest of A 
+    @inbounds for i in 1:length(S)
+        A[i] = zero(T)
+    end
+    # Copy the structural nonzeros from S to M using 
+    # the linear indices (to work when size(M)!=size(S))
+    num_rows = size(S,1)
+    colptr = getcolptr(S)
+    rowval = getrowval(S)
+    nzval = getnzval(S)
+    for col in 1:size(S, 2)
+        for i in nzrange(S, col)
+            row = rowval[i]
+            val = nzval[i]
+            linear_index = num_rows*(col-1)+row
```

Comment:
Not sure it makes a big difference, but do you want to hoist out `linear_index = num_rows*(col-1)` out of the inner loop and write into `A[linear_index+row]`? (Or actually start with `linear_index = 0` outside of all loops and then `linear_index += num_rows` in the `col` loop? And perhaps mark the `i`-loop as `@inbounds`?

Reply:
Thanks, the accumulation part is for sure nice! I'll update with that.

For the inbounds, I'm not sure which checks will be required in the outer loop? 
E.g., is `nzrange` safe for indexing `rowval` and `nzval`? And further, I don't understand/know if one can rely on 
`rowval[nzrange(S,col)]` to be sorted (and only check first and last value).

Reply:
Now I also moved it to the other specialized copyto! methods (I think it is better to put them all together or?)
We don't need any `unalias` right? I could not construct a case to make, e.g. `nzvals` and `A` aliased.